pages/[...slug].tsx

```diff
@@ -50,27 +50,35 @@ export default function PostPage({ allDocs, nav, toc, data, source }: PostPagePr
   )
 }
 
+interface PostPageProps extends Partial<PostProps> {
+  redirect?: string
+}
+
+export default function PostPage({ redirect, ...rest }: PostPageProps) {
+  const router = useRouter()
+
+  useLayoutEffect(() => {
+    if (redirect) return void router.push(redirect)
+  }, [redirect, router])
+
+  return redirect ? null : <Post {...rest} />
+}
+
```

Comment:
Maybe we can serve the page specified as `redirect` instead? Ideally, we can change the URL as well without consequence. Looking into [*shallow routing*](https://nextjs.org/docs/routing/shallow-routing).